types/address_test.go

```diff
@@ -507,15 +565,234 @@ func (s *addressTestSuite) TestGetFromBech32() {
 	s.Require().Equal("invalid Bech32 prefix; expected x, got cosmos", err.Error())
 }
 
+func (s *addressTestSuite) TestMustValAddressFromBech32() {
+	valAddress1 := types.ValAddress([]byte{0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19})
+	valAddress2 := types.MustValAddressFromBech32(valAddress1.String())
+
+	s.Require().Equal(valAddress1, valAddress2)
+}
+
+func (s *addressTestSuite) TestMustValAddressFromBech32Panic() {
+	s.Require().Panics(func() {
+		types.MustValAddressFromBech32("")
+	})
+}
+
+func (s *addressTestSuite) TestGetBech32PrefixAccPub() {
+	actual := types.GetBech32PrefixAccPub("")
+	s.Require().Equal("pub", actual)
+
+	actual = types.GetBech32PrefixAccPub("cosmos")
+	s.Require().Equal("cosmospub", actual)
+}
+
+func (s *addressTestSuite) TestGetBech32PrefixValAddress() {
+	actual := types.GetBech32PrefixValAddr("")
+	s.Require().Equal("valoper", actual)
+
+	actual = types.GetBech32PrefixValAddr("cosmos1")
+	s.Require().Equal("cosmos1valoper", actual)
+}
+
+func (s *addressTestSuite) TestGetBech32PrefixValPub() {
+	actual := types.GetBech32PrefixValPub("")
+	s.Require().Equal("valoperpub", actual)
+
+	actual = types.GetBech32PrefixValPub("cosmos2")
+	s.Require().Equal("cosmos2valoperpub", actual)
+}
+
+func (s *addressTestSuite) TestGetBech32PrefixConsAddr() {
+	actual := types.GetBech32PrefixConsAddr("")
+	s.Require().Equal("valcons", actual)
+
+	actual = types.GetBech32PrefixConsAddr("cosmos3")
+	s.Require().Equal("cosmos3valcons", actual)
+}
+
+func (s *addressTestSuite) TestGetBech32PrefixConsPub() {
+	actual := types.GetBech32PrefixConsPub("")
+	s.Require().Equal("valconspub", actual)
+
+	actual = types.GetBech32PrefixConsPub("cosmos4")
+	s.Require().Equal("cosmos4valconspub", actual)
+}
+
 func (s *addressTestSuite) TestMustAccAddressFromBech32() {
-	bech32PrefixValAddr := types.GetConfig().GetBech32ValidatorAddrPrefix()
+	accAddress1 := types.AccAddress([]byte{0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19})
+	accAddress2 := types.MustAccAddressFromBech32(accAddress1.String())
+	s.Require().Equal(accAddress1, accAddress2)
```

Comment:
I would use the full bech32 string here to ensure that the decoding works.  Otherwise you can not really know what encoding is used.
```go
func (s *addressTestSuite) TestMustAccAddressFromBech32() {
	src := types.MustAccAddressFromBech32("cosmos1qqqsyqcyq5rqwzqfpg9scrgwpugpzysnrk363e")
	exp := types.AccAddress([]byte{0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19})
	s.Assert().Equal(exp, src)
}
```

Reply:
Done

Reply:
@aple now I remember why I did this. The prefix is always different randomly generated string. If I use "cosmos" as a prefix the test always fail when I run all the tests.